saleor/discount/models.py

```diff
@@ -384,6 +384,7 @@ class PromotionRule(models.Model):
         blank=True,
     )
     old_channel_listing_id = models.IntegerField(blank=True, null=True, unique=True)
+    recalculate_rule = models.BooleanField(default=False)
```

Comment:
I think we should find out a better name for this field. For example to mark that product `search_document` needs recalculation we have a field `search_index_dirty`. Maybe we should use something like `variants_dirty`? We can discuss the field name with the team as well